services/virtual-assistant/src/virtual-assistant/api_types.py

```diff
@@ -1,13 +1,103 @@
 from pydantic import BaseModel
-from typing import Optional
+from typing import Optional, List
 
 
-class TalkRequestMetadata(BaseModel):
-    session_id: Optional[str] = None
-    # account: str
-    # org_id: str
+class TalkRequestInputIntents(BaseModel):
+    intent: str
+    confidence: Optional[float] = None
+    skill: Optional[str] = None
+
+
+class TalkRequestInputEntities(BaseModel):
+    entity: str
+    location: Optional[List[int]] = []
+    value: str
+    confidence: Optional[float] = None
+    groups: Optional[dict] = {}
+    interpretation: Optional[dict] = {}
+    alternatives: Optional[dict] = {}
+    role: Optional[dict] = {}
+    skill: Optional[str] = {}
+
+
+class TalkRequestInputAttachments(BaseModel):
+    url: str
+    media_type: Optional[str] = None
+
+
+class TalkRequestInputAnalytics(BaseModel):
+    browser: Optional[str] = None
+    device: Optional[str] = None
+    page_url: Optional[str] = None
+
+
+class TalkInput(BaseModel):
+    message_type: Optional[str] = None
+    text: Optional[str] = None
+    intents: Optional[List[TalkRequestInputIntents]] = None
+    entities: Optional[List[TalkRequestInputEntities]] = None
+    suggestion_id: Optional[str] = None
+    attachments: Optional[TalkRequestInputAttachments] = None
+    analytics: Optional[TalkRequestInputAnalytics] = None
+    options: Optional[dict] = None
```

Comment:
These types looks way too complex that what we really need. IMO we shouldn't try to be a proxy of watson, but provide a simple API and we do the heavy lifting.

Lets try to simplify this going forward, we can leave space for extending as more feature are required but lets try to only add what we are going to be using initially (i.e. we can omit extra attributes for nice to have)